### DISCUSSION TITLE
Enhancement: True/Exclusive Fullscreen Mode

### DISCUSSION THREAD

**Discussion by sentrixcoin** [FunkinCrew/Funkin#1607] [Ideas]
Adding this to the game would reduce input lag and increase overall performance, Rhythm games require the most accurate input processing possible as you need to input as quickly as possible. Exclusive Fullscreen renders the game directly on your screen instead of going through windows DWM (Desktop Windows Manager), it's the rawest and fastest possible rendering you could achieve with a game. If you need help implementing or have any questions let me know.

**Comment by TentaRJ** [comment]
> Adding this to the game would reduce input lag and increase overall performance, Rhythm games require the most accurate input processing possible as you need to input as quickly as possible. Exclusive Fullscreen renders the game directly on your screen instead of going through windows DWM (Desktop Windows Manager), it's the rawest and fastest possible rendering you could achieve with a game. If you need help implementing or have any questions let me know.

All you need to do to make the game fullscreen is to go into project.xml and change the fullscreen value of your build from 'false' to 'true'.

```haxe
<!-- ____________________________ Window Settings ___________________________ -->

	<!--These window settings apply to all targets-->
	<window width="1280" height="720" fps="60" background="#000000" hardware="true" vsync="false" />

	<!--HTML5-specific-->
	<window if="html5" resizable="true" />

	<!--Desktop-specific-->
	<window if="desktop" orientation="landscape" fullscreen="false" resizable="true" vsync="false"/>

	<!--Mobile-specific-->
	<window if="mobile" orientation="landscape" fullscreen="true" width="0" height="0" />

	<!--Switch-specific-->
	<window if="switch" orientation="landscape" fullscreen="true" width="0" height="0" resizable="true" />
```

**Comment by T0prakk** [comment]
I really like this, as of my laptop is trash and I wanna play FNF, I mostly I am able to play games in low screens, thanks to my laptop. (except for old games and such)

**Comment by vinjcc** [comment]
> All you need to do to make the game fullscreen is to go into project.xml and change the fullscreen value of your build from 'false' to 'true'.

you can also do `lime test linux -debug --window-fullscreen=true`
e: or whatever platform you're compiling for

**Comment by sentrixcoin** [comment]
> > All you need to do to make the game fullscreen is to go into project.xml and change the fullscreen value of your build from 'false' to 'true'.
> 
> you can also do `lime test linux -debug --window-fullscreen=true`
> e: or whatever platform you're compiling for

Have you guys tested this? I have a sneaking suspicion that it will more than likely be some variation of Windowed Fullscreen, not True/Exclusive Fullscreen. I unfortunately can't compile this myself to test it.

**Comment by suprstarrd** [comment]
@sentrixcoin Will test later today

**Comment by sentrixcoin** [comment]
> @sentrixcoin Will test later today

Sounds good! Also, I'm assuming that you're on Windows 10. If you are make sure to right click the game exe file, go to properties, compatibility and then check the "Disable Fullscreen Optimizations" button. Really important that you do this when testing.

**Comment by sentrixcoin** [comment]
@jbmagination Hey, did you ever test it? Sorry to bother you!

**Comment by suprstarrd** [comment]
> @jbmagination Hey, did you ever test it? Sorry to bother you!

Ah, no! Thanks for the reminder, will go ahead and do it right now.

**Comment by GStudiosX2** [comment]
He can't do that as all the rendering stuff is done by haxeflixel and plausibly not possible
